packages/test-sync/src/androidAndroidTest/kotlin/io/realm/kotlin/test/mongodb/shared/FunctionsTests.kt

```diff
@@ -0,0 +1,623 @@
+                    assertTypeOfFirstArgFunction(org.mongodb.kbson.BsonObjectId())
+                }
+                BsonType.BOOLEAN -> {
+                    assertTrue(functions.callBlocking(FIRST_ARG_FUNCTION.name, true))
+                    assertTypeOfFirstArgFunction(BsonBoolean(true))
+                }
+                BsonType.INT32 -> {
+                    assertEquals(
+                        32,
+                        functions.callBlocking<Int>(FIRST_ARG_FUNCTION.name, 32).toInt()
+                    )
+                    assertEquals(
+                        32,
+                        functions.callBlocking<Int>(FIRST_ARG_FUNCTION.name, 32L).toInt()
+                    )
+                    assertTypeOfFirstArgFunction(BsonInt32(32))
+                }
+                BsonType.INT64 -> {
+                    assertEquals(
+                        32L,
+                        functions.callBlocking<Long>(FIRST_ARG_FUNCTION.name, 32L).toLong()
+                    )
+                    assertEquals(
+                        32L,
+                        functions.callBlocking<Long>(FIRST_ARG_FUNCTION.name, 32).toLong()
+                    )
+                    assertTypeOfFirstArgFunction(BsonInt64(32))
+                }
+                BsonType.DECIMAL128 -> {
+                    assertTypeOfFirstArgFunction(BsonDecimal128("32"))
+                }
+                BsonType.DOCUMENT -> {
+                    val map = mapOf("foo" to 5)
+                    val document = BsonDocument(mapOf("foo" to BsonInt32(5)))
+
+                    assertEquals(
+                        document,
+                        functions.callBlocking<BsonDocument>(FIRST_ARG_FUNCTION.name, map)
+                    )
+                    assertEquals(
+                        document,
+                        functions.callBlocking<BsonDocument>(FIRST_ARG_FUNCTION.name, document)
+                    )
+
+                    var documents = arrayOf(BsonDocument(), BsonDocument())
+                    assertEquals(
+                        documents[0],
+                        functions.callBlocking<BsonDocument>(FIRST_ARG_FUNCTION.name, *documents)
+                    )
+
+                    documents = arrayOf(
+                        BsonDocument("KEY", BsonString("VALUE")),
+                        BsonDocument("KEY", BsonString("VALUE")),
+                        BsonDocument("KEY", BsonString("VALUE"))
+                    )
+                    assertEquals(
+                        documents[0],
+                        functions.callBlocking<BsonDocument>(FIRST_ARG_FUNCTION.name, *documents)
+                    )
+                }
+                BsonType.DATE_TIME -> {
+                    // JVM and Darwing platform's RealmInstant have better precission than BsonDateTime
+                    // we create a RealmInstant with loose of precision
```

Comment:
Not sure I get the comment??

Reply:
`RealmInstant` has nanoseconds precision while `BsonDateTime` has milliseconds. I just wrote it in a funny way 🫣